src/coreclr/jit/targetamd64.h

```diff
@@ -390,7 +390,7 @@
   // The following defines are useful for iterating a regNumber
   #define REG_FIRST                REG_EAX
   #define REG_INT_FIRST            REG_EAX
-  #define REG_INT_LAST             REG_R23
```

Comment:
should this be `REG_R15` by default and then you update it to `REG_R31` if APX is supported in `compInitOptions()` `regIntLast = REG_R31`?

Reply:
Having it as `REG_R31` and changing it `REG_R15` works better I think because we have some static asserts that uses REG_INT_LAST for sanity checks. That's the reason why we decided to go with this in the original PR where I added `REG_R16` - `REG_R23`